File: Sorting/SelectionSort.java
public class SelectionSort {

	public static void sort(int[] arr) {
		for (int i = 0; i < arr.length; i++) {
			int min = i;
			for (int j = i + 1; j < arr.length; j++) {
				/* find local min */
				if (arr[j] < arr[min]) {
					min = j;
				}
			}
			swap(arr, i, min);
		}
	}

	private static void swap(int[] arr, int i, int j) {
		int temp = arr[i];
		arr[i] = arr[j];
		arr[j] = temp;
	}
}

File: Sorting/SelectionSort.js
var selectionSort = function(arr) {
  for(var i = 0; i < arr.length; i++){
    //set low to the current value of i
    var low = i;
    for(var j = i+1; j < arr.length; j++){
      if(arr[j] < arr[low]){
       low = j;
      }
    }
    var temp = arr[i];
    arr[i] = arr[low];
    arr[low] = temp;
  }
  return arr;
};
